rllib/core/models/torch/encoder.py

```diff
@@ -0,0 +1,198 @@
+                },
+                SampleBatch.SEQ_LENS: None,
+            }
+        )
+
+    @override(Model)
+    def get_output_spec(self) -> Union[Spec, None]:
+        return SpecDict(
+            {
+                ENCODER_OUT: TorchTensorSpec("bxt, h", h=self.config.output_dim),
+                STATE_OUT: {
+                    "h": TorchTensorSpec(
+                        "b, l, h", h=self.config.hidden_dim, l=self.config.num_layers
+                    ),
+                    "c": TorchTensorSpec(
+                        "b, l, h", h=self.config.hidden_dim, l=self.config.num_layers
+                    ),
+                },
+            }
+        )
+
+    @override(Model)
+    def get_initial_state(self):
+        config = self.config
+        return {
+            "h": torch.zeros(config.num_layers, config.hidden_dim),
+            "c": torch.zeros(config.num_layers, config.hidden_dim),
+        }
+
+    @override(Model)
+    def _forward(self, inputs: NestedDict, **kwargs) -> NestedDict:
+        x = inputs[SampleBatch.OBS]
+        states = inputs[STATE_IN]
+        # states are batch-first when coming in
+        states = tree.map_structure(lambda x: x.transpose(0, 1), states)
+
+        x = add_time_dimension(
+            x,
+            seq_lens=inputs[SampleBatch.SEQ_LENS],
+            framework="torch",
+            time_major=not self.config.batch_first,
+        )
+        states_o = {}
+        x, (states_o["h"], states_o["c"]) = self.lstm(x, (states["h"], states["c"]))
+
+        x = self.linear(x)
+        x = x.view(-1, x.shape[-1])
+
+        return {
+            ENCODER_OUT: x,
+            STATE_OUT: tree.map_structure(lambda x: x.transpose(0, 1), states_o),
+        }
+
+
+class TorchIdentityEncoder(TorchModel, Encoder):
+    """An encoder that does nothing but passing on inputs.
+
+    We use this so that we avoid having many if/else statements in the RLModule.
+    """
+
+    def __init__(self, config: ModelConfig) -> None:
+        TorchModel.__init__(self, config)
+        Encoder.__init__(self, config)
+
+    @override(Model)
+    def get_input_spec(self) -> Union[Spec, None]:
+        return SpecDict(
+            # Use the output dim as input dim because identity.
+            {
+                SampleBatch.OBS: TorchTensorSpec("b, h", h=self.config.output_dim),
+                STATE_IN: None,
+                SampleBatch.SEQ_LENS: None,
+            }
+        )
+
+    @override(Model)
+    def get_output_spec(self) -> Union[Spec, None]:
+        return SpecDict(
+            {
+                ENCODER_OUT: TorchTensorSpec("b, h", h=self.config.output_dim),
+                STATE_OUT: None,
+            }
+        )
+
+    @override(Model)
+    def _forward(self, inputs: NestedDict, **kwargs) -> NestedDict:
+        return NestedDict(
+            {ENCODER_OUT: inputs[SampleBatch.OBS], STATE_OUT: inputs[STATE_IN]}
+        )
+
+
+class TorchActorCriticEncoder(TorchModel, ActorCriticEncoder):
```

Comment:
For example this should call 

```
TorchModel().__init__()
ActorCriticEncoder().__init__()
```

Reply:
what's wrong with this?

Reply:
See https://github.com/ray-project/ray/pull/32453#discussion_r1106566241